### ISSUE TITLE
Webs server display of "Contactors controlled by emulator, state:" not working correctly

### ISSUE THREAD

**Issue by davidhgriffiths** [dalathegreat/Battery-Emulator#679]
### Expected Behavior

When the contactors open after an inverter request, the web server should display the OFF state.

### Actual Behavior

When the contactors open after an inverter request the web server continues to display the  ON state

I think the bug is caused by the absence of the following line at about line 128 of webserver.cpp

```
datalayer.system.status.contactors_engaged = false;
```

Currently `datalayer.system.status.contactors_engaged` only gets set to false if contractor opening is requested from the web page, not if it is requested by the inverter or estop

**Comment by mvgalen** [comment]
Just so that I understand the issue correct: You mean that currently if an inverter request opening of contactors it does not show on the webpage? For which inverter have you seen this?

**Comment by davidhgriffiths** [comment]
Yes your understanding is correct - if the inverter requests opening of the contractors the contractors open but the web page continues to show them closed.  The inverter is a Solax X3 G4. With the change I suggested it works correctly.

**Comment by dalathegreat** [comment]
Thanks for the report @davidhgriffiths , I have issued a fix for this in PR https://github.com/dalathegreat/Battery-Emulator/pull/812

**Comment by dalathegreat** [comment]
Closing as completed, now in main, target release 8.3.0 🥳  Thanks again for the report!
